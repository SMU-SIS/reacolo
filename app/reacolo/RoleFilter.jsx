import React from 'react';
import ComponentFilter from './ComponentFilter';

const arrayIncludes = Array.prototype.includes;

export default function RoleFilter(props) {
  let shouldRender;
  if (typeof props.target === 'string') {
    shouldRender = props.target === props.context.role;
  } else {
    shouldRender = arrayIncludes.call(props.target, props.context.role);
  }
  return (
    <ComponentFilter rendered={shouldRender}>
      {props.children}
    </ComponentFilter>
  );
}

RoleFilter.propTypes = {
  target: React.PropTypes.oneOfType([
    React.PropTypes.string,
    React.PropTypes.arrayOf(React.PropTypes.string)
  ]).isRequired,
  children: React.PropTypes.element.isRequired,
  context: React.PropTypes.shape({
    role: React.PropTypes.string.isRequired
  }).isRequired
};
