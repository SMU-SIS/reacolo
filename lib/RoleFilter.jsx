import React from 'react';
import ComponentFilter from './ComponentFilter';

const filter = (targetOrTargets, role) => {
  if (typeof target === 'string') {
    return targetOrTargets === role;
  }
  return targetOrTargets.includes(role);
};

const RoleFilter = props => (
  <ComponentFilter rendered={filter(props.target, props.context.role)}>
    { props.children }
  </ComponentFilter>
);

RoleFilter.propTypes = {
  target: React.PropTypes.oneOfType([
    React.PropTypes.string,
    React.PropTypes.arrayOf(React.PropTypes.string)
  ]).isRequired,
  children: React.PropTypes.node.isRequired,
  context: React.PropTypes.shape({
    role: React.PropTypes.string
  }).isRequired
};

export default RoleFilter;
