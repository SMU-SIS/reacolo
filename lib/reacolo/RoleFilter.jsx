import React from 'react';
import ValueFilter from './ValueFilter';

const RoleFilter = props => (
  <ValueFilter value={props.context.role} target={props.target}>
    { props.children }
  </ValueFilter>
);

RoleFilter.propTypes = {
  target: React.PropTypes.oneOfType([
    React.PropTypes.string,
    React.PropTypes.arrayOf(React.PropTypes.string)
  ]).isRequired,
  children: React.PropTypes.node.isRequired,
  context: React.PropTypes.shape({
    role: React.PropTypes.string
  }).isRequired
};

export default RoleFilter;
