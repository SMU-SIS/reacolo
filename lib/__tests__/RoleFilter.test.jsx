import React from 'react';
import renderer from 'react-test-renderer';
import RoleFilter from '../reacolo/RoleFilter';

test('RoleFilter renders if the context\'s role is the target role.', () => {
  const component = renderer.create(
    <div id="container">
      <RoleFilter target="test" context={{ role: 'test' }} >
        <div id="child" />
      </RoleFilter>
    </div>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('RoleFilter does not render if the context\'s role is not the target role.', () => {
  const component = renderer.create(
    <div id="container">
      <RoleFilter target="test" context={{ role: 'something-else' }} >
        <div id="child" />
      </RoleFilter>
    </div>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('RoleFilter renders if the context\'s role same is one of the targets.', () => {
  const targets = ['target1', 'target2', 'target3'];
  const context = { role: 'target2' };
  const component = renderer.create(
    <div id="container">
      <RoleFilter target={targets} context={context} >
        <div id="child" />
      </RoleFilter>
    </div>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('RoleFilter does not render if the context\'s role is not one of the targets.', () => {
  const targets = ['target1', 'target2', 'target3'];
  const context = { role: 'something-else' };
  const component = renderer.create(
    <div id="container">
      <RoleFilter target={targets} context={context} >
        <div id="child" />
      </RoleFilter>
    </div>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
