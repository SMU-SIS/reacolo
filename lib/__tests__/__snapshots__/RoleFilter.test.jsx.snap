exports[`test RoleFilter does not render if the context's role is not one of the targets. 1`] = `
<div
  id="container" />
`;

exports[`test RoleFilter does not render if the context's role is not the target role. 1`] = `
<div
  id="container" />
`;

exports[`test RoleFilter renders if the context's role is the target role. 1`] = `
<div
  id="container">
  <div>
    <div
      id="child" />
  </div>
</div>
`;

exports[`test RoleFilter renders if the context's role same is one of the targets. 1`] = `
<div
  id="container">
  <div>
    <div
      id="child" />
  </div>
</div>
`;
